src/main/java/org/example/services/Services.java

```diff
@@ -25,59 +24,51 @@ public static boolean isPerfectNumber(int artifactCode) {
                 sum += i;
             }
         }
-        if (sum == artifactCode) {
+
+        if (artifactCode == 0) {
+            return isPerfect;
+        } else if (sum == artifactCode) {
             isPerfect = true;
         }
         return isPerfect;
     }
 
-    public static boolean isPrime(int artifact) {
+    @Override
+    public boolean isPrime(int artifact) {
         if (artifact <= 1) {
             return false;
         }
 
-        for (int i = 2; i < Math.sqrt(artifact); i++) {
+        for (int i = 2; i < artifact; i++) {
             if (artifact % i == 0) {
                 return false;
             }
         }
         return true;
     }
 
-    public static boolean isTrap(int artifact) {
-        return isSumDivBy3(artifact) && isEven(artifact);
-    }
-
-    private static boolean isSumDivBy3(int artifact) {
-        int sum = 0;
-        int num = artifact;
-
-        while (num != 0) {
-            sum += num % 10;
-            num /= 10;
-        }
-
-        return sum % 3 == 0;
-    }
-
-    private static boolean isEven(int artifact) {
-        return artifact % 2 == 0;
+    @Override
+    public boolean isTrap(int artifact) {
+        return isSumDivBy3(artifact) && artifact % 2 == 0;
     }
 
-    public static boolean winProbability(PlayerStatus opponent, int health, int score) {
+    @Override
+    public boolean myWinProbability(PlayerStatus opponent, int health, int score) {
         var opponentWinProb = (3 * opponent.getHealth() + opponent.getScore() / 1000) / 4;
         var myWinProb = (3 * health + score / 1000) / 4;
 
-        return myWinProb > opponentWinProb;
+        return myWinProb >= opponentWinProb;
     }
 
-    public static double getDistance(PlayerStatus opponent, double posX, double posY) {
+    @Override
+    public double getDistance(PlayerStatus opponent, double posX, double posY) {
```

Comment:
Refactor this to use as parameters two instances of PlayerStatus instead.

Reply:
Solved